controllers/result.go

```diff
@@ -129,6 +133,20 @@ func (rc *ResultController) getResultID() (int64, error) {
 	return resultID, nil
 }
 
+func (rc *ResultController) queryFromParams() map[string]interface{} {
+	searchKeyword := rc.GetString("keyword")
+
+	query := map[string]interface{}{
+		"user_id":            rc.CurrentUser.Id,
+		"order":              "-created_at",
+		"keyword__icontains": searchKeyword,
+	}
+
+	rc.Data["keyword"] = searchKeyword
```

Comment:
As `queryFromParams` function's purpose is to generate the query, I think we can get rid out of here this is not related to generating query rather is setting data for view. I think we can return `searchKeyword ` from this function and add `rc.Data["keyword"] = searchKeyword` near with `rc.Data["resultSets"] = resultSets` in the `List ` function. What do you think?